lib/api/multiObjectDelete.js

```diff
@@ -191,166 +248,187 @@ function getObjMetadataAndDelete(authInfo, canonicalID, request,
     let numOfObjectsRemoved = 0;
     const skipError = new Error('skip');
     const objectLockedError = new Error('object locked');
+    let deleteFromStorage = [];
+    const initialStep = initializeMultiObjectDeleteWithBatchingSupport(bucketName, inPlay, log);
```

Comment:
the callback initialized from the function makes the code harder to follow/read, and I don't see the benefit of the functional approach here : we may as well do what needs to be done in the waterfall, there be one "branch" anyway to handle both caching/no-caching...

Reply:
I reworked the function call to directly return the callback result: https://github.com/scality/cloudserver/pull/5195/commits/21d22ad1832687ed188f284dcafeaf6ae1c4f285#diff-e8319ec288d282012f389d259d6468a0922b91ec7cc52f2b8c15b01ff3ba9c2aR201 && https://github.com/scality/cloudserver/pull/5195/commits/21d22ad1832687ed188f284dcafeaf6ae1c4f285#diff-e8319ec288d282012f389d259d6468a0922b91ec7cc52f2b8c15b01ff3ba9c2aR254